### ISSUE TITLE
CA1806 "Do not ignore method results" in Utility.cs:135

### ISSUE THREAD

**Issue by ariccio** [EWSoftware/VSSpellChecker#29]
[In `Utility.cs`:](https://github.com/EWSoftware/VSSpellChecker/blob/master/Source/VSSpellChecker/Utility.cs#L135)

```
                        if(persistFileFormat != null)
                        {
                            persistFileFormat.GetCurFile(out ppzsFilename, out pnFormatIndex);

                            if(!String.IsNullOrEmpty(ppzsFilename))
                                return ppzsFilename;
                        }

```
Code Analysis complains:

> `Utility.GetFilename(this ITextBuffer)` calls `IPersistFileFormat.GetCurFile(out string, out uint)` but does not use the `HRESULT` or error code that the method returns. This could lead to unexpected behavior in error conditions or low-resource situations. Use the result in a conditional statement, assign the result to a variable, or pass it as an argument to another method.

I'm not a `C#` developer, so I don't know what the proper way of checking this `HRESULT` is, but I can't imagine that ignoring it is a good idea.

**Comment by EWSoftware** [comment]
It returns S_OK if it succeeds or an error code if it fails.  If it fails, its doubtful it'll return anything other than null for the filename so the null or empty check will suffice.  It's not allocating resources so there's nothing to free.  I typically suppress these warnings after review unless there's a good reason to check the return code.
